File: lib/models/destiny_item_set_action_request.dart
class DestinyItemSetActionRequest{
	List<String> itemIds;
	String characterId;
	int membershipType;
	DestinyItemSetActionRequest(
		List<String> this.itemIds,
		String this.characterId,
		int this.membershipType,
	);

	static DestinyItemSetActionRequest fromMap(Map<String, dynamic> data){
		if(data == null) {
			return null;
		};
		return new DestinyItemSetActionRequest(
				data['itemIds'],
				data['characterId'],
				data['membershipType'],
		);
	}

	static List<DestinyItemSetActionRequest> fromList(List<dynamic> data){
		if(data == null) {
			return null;
		};
		List<DestinyItemSetActionRequest> list = new List();
    data.forEach((item) {
      list.add(DestinyItemSetActionRequest.fromMap(item));
    });
    return list;
	}

	Map<String, dynamic> toMap(){
		Map<String, dynamic> data = new Map();
			data['itemIds'] = this.itemIds;
			data['characterId'] = this.characterId;
			data['membershipType'] = this.membershipType;
		return data;
	}
}

File: lib/src/models/DestinyItemSetActionRequest.cs
using System.Runtime.Serialization;
using System.Collections.Generic;

namespace BungieNetApi.Model {
	[DataContract]
	public class DestinyItemSetActionRequest{

		[DataMember(Name="itemIds", EmitDefaultValue=false)]
		public List<long> ItemIds { get; set; }

		[DataMember(Name="characterId", EmitDefaultValue=false)]
		public long CharacterId { get; set; }

		[DataMember(Name="membershipType", EmitDefaultValue=false)]
		public BungieMembershipType MembershipType { get; set; }


		public override bool Equals(object input)
        {
            return this.Equals(input as DestinyItemSetActionRequest);
        }

		public bool Equals(DestinyItemSetActionRequest input)
		{
			if (input == null) return false;

			return
				(
                    ItemIds == input.ItemIds ||
                    (ItemIds != null && ItemIds.Equals(input.ItemIds))
                ) &&
				(
                    CharacterId == input.CharacterId ||
                    (CharacterId != null && CharacterId.Equals(input.CharacterId))
                ) &&
				(
                    MembershipType == input.MembershipType ||
                    (MembershipType != null && MembershipType.Equals(input.MembershipType))
                ) ;
		}
	}
}

